### PR TITLE
High score backend

### PR DESCRIPTION
**Pull request by john-li7**
- GET router for '/high-score' first verifies the user has a cookie with a matching active session via sessionController.isLoggedIn. I created the scoreController.getHighScore middleware function that checks if res.locals.cookieSessionMatch is true. If so, a query is made to the high_score table to get the high_score column for records where users_id is equal to the request cookie named ssid which is created during successful signup/login. The query result is stored into res.locals.highScore. End of route, if cookieSessionMatch is true, will return res.locals. If false, return 'Invalid session'.
- PUT router for '/high-score' first verifies the user has a cookie with matching active session via sessionController.isLoggedIn and then runs the scoreController.getHighScore function created above. I created the scoreController.updateHighScore middleware function to be next in the chain. The function checks for res.locals.cookieSessionMatch. If true, it checks if the req.body.score is greater than res.locals.highScore. If so, a query will be made to update the high_score in the high_score table where the users_id is equal to the request cookie named ssid. After this function, scoreController.getHighScore is ran again to get the new high score. The end of the router is the same as app.get('/high-score') above.
- Updated userController.createUser so a new record in high_score will be created on successful signup. The new record is created with the newly created user's _id.

### CODE REVIEW

File: server/server.js [hunk 1]
```diff
@@ -69,6 +70,30 @@ app.get(
...
 );
 
+app.get(
+  '/high-score',
+  sessionController.isLoggedIn,
+  scoreController.getHighScore,
+  (req, res) => {
+    if (!res.locals.cookieSessionMatch)
+      return res.status(200).json('Invalid session');
+    return res.status(200).json(res.locals);
```
Comment (unknownbreaker) [inline server/server.js:80]: Wondering if the two `returns` should return an object of the same structure. The successful session match returns an object with a key of highScore, but the unsuccessful (which is actually still successful) is just a string. If you made them both the same structure, then could be easier to reason about on the frontend.

What do you think?

File: server/server.js [hunk 2]
```diff
@@ -69,6 +70,30 @@ app.get(
...
+app.put(
+  '/high-score',
+  sessionController.isLoggedIn,
+  scoreController.getHighScore,
+  scoreController.updateHighScore,
+  scoreController.getHighScore,
+  (req, res) => {
+    if (!res.locals.cookieSessionMatch)
+      return res.status(200).json('Invalid session');
+    return res.status(200).json(res.locals);
```
Comment (unknownbreaker) [inline server/server.js:93]: Wondering if the two `returns` should return an object of the same structure. The successful session match returns an object with a key of highScore, but the unsuccessful (which is actually still successful) is just a string. If you made them both the same structure, then could be easier to reason about on the frontend.

What do you think?

**Review by unknownbreaker — CHANGES_REQUESTED** [review]
Sent back a couple comments for review.

File: server/server.js (hunk 1 above)
Comment (john-li7) [reply server/server.js:80]: An unsuccessful cookieSessionMatch means there is no valid session for the user. Ideally, this should redirect the user to the sign in page. However, didn't want to add this yet until front-end was able to send GET request to '/high-score' to test how it plays out.

File: server/server.js (hunk 2 above)
Comment (john-li7) [reply server/server.js:93]: Same as above.

File: server/server.js (hunk 1 above)
Comment (unknownbreaker) [reply server/server.js:80]: Sounds good.

File: server/server.js (hunk 2 above)
Comment (unknownbreaker) [reply server/server.js:93]: Sounds good.

**Review by unknownbreaker — APPROVED** [review]

**Pull request merged**